MSAL/src/MSALAccountId.m

```diff
@@ -78,8 +76,16 @@ - (BOOL)isEqualToItem:(MSALAccountId *)accountId
 {
     BOOL result = YES;
     result &= (!self.identifier && !accountId.identifier) || [self.identifier isEqualToString:accountId.identifier];
-    result &= (!self.objectId && !accountId.objectId) || [self.objectId isEqualToString:accountId.objectId];
-    result &= (!self.tenantId && !accountId.tenantId) || [self.tenantId isEqualToString:accountId.tenantId];
+    
+    if (self.objectId && accountId.objectId)
```

Comment:
Can you please explain the reason for this change. This test will ignore the case when either self.objectId or accountId.objectId is nil.

Reply:
objectId and tenantId are optional fields.
identifier is the mandatory field. 
Two account IDs should be equal if their identifiers are equal. If one account id simply doesn't have objectId/tenantId filled in, it doesn't mean it's not equivalent to the other account id.